packages/flutter/test/painting/box_painter_test.dart

```diff
@@ -107,7 +109,284 @@ void main() {
     expect(shadowList, equals(<BoxShadow>[shadow4, shadow1.scale(0.5)]));
   });
 
+  test('BoxShadow BlurStyle test', () {
+    const BoxShadow shadow1 = BoxShadow(blurRadius: 4.0);
+    const BoxShadow shadow2 = BoxShadow(blurRadius: 4.0, blurStyle: BlurStyle.outer);
+    final BoxShadow shadow3 = BoxShadow.lerp(shadow1, null, 0.25)!;
+    final BoxShadow shadow4 = BoxShadow.lerp(null, shadow1, 0.25)!;
+    final BoxShadow shadow5 = BoxShadow.lerp(shadow1, shadow2, 0.25)!;
+    final BoxShadow shadow6 = BoxShadow.lerp(const BoxShadow(blurStyle: BlurStyle.solid), shadow2, 0.25)!;
+
+    expect(shadow1.blurStyle, equals(BlurStyle.normal));
+    expect(shadow2.blurStyle, equals(BlurStyle.outer));
+    expect(shadow3.blurStyle, equals(BlurStyle.normal));
+    expect(shadow4.blurStyle, equals(BlurStyle.normal));
+    expect(shadow5.blurStyle, equals(BlurStyle.outer));
+    expect(shadow6.blurStyle, equals(BlurStyle.solid));
+
+    List<BoxShadow> shadowList = BoxShadow.lerpList(<BoxShadow>[shadow2, shadow1], <BoxShadow>[shadow3], 0.5)!;
+    expect(shadowList[0].blurStyle, equals(BlurStyle.outer));
+    expect(shadowList[1].blurStyle, equals(BlurStyle.normal));
+
+    shadowList = BoxShadow.lerpList(<BoxShadow>[shadow6], <BoxShadow>[shadow3, shadow1], 0.5)!;
+    expect(shadowList[0].blurStyle, equals(BlurStyle.solid));
+    expect(shadowList[1].blurStyle, equals(BlurStyle.normal));
+
+    shadowList = BoxShadow.lerpList(<BoxShadow>[shadow3], <BoxShadow>[shadow6, shadow1], 0.5)!;
+    expect(shadowList[0].blurStyle, equals(BlurStyle.solid));
+    expect(shadowList[1].blurStyle, equals(BlurStyle.normal));
+
+    shadowList = BoxShadow.lerpList(<BoxShadow>[shadow3], <BoxShadow>[shadow2, shadow1], 0.5)!;
+    expect(shadowList[0].blurStyle, equals(BlurStyle.outer));
+    expect(shadowList[1].blurStyle, equals(BlurStyle.normal));
+  });
+
   test('BoxShadow toString test', () {
-    expect(const BoxShadow(blurRadius: 4.0).toString(), equals('BoxShadow(Color(0xff000000), Offset(0.0, 0.0), 4.0, 0.0)'));
+    expect(const BoxShadow(blurRadius: 4.0).toString(), equals('BoxShadow(Color(0xff000000), Offset(0.0, 0.0), 4.0, 0.0), BlurStyle.normal'));
+    expect(const BoxShadow(blurRadius: 4.0, blurStyle: BlurStyle.solid).toString(), equals('BoxShadow(Color(0xff000000), Offset(0.0, 0.0), 4.0, 0.0), BlurStyle.solid'));
+  });
+
+  testWidgets('BoxShadow BoxStyle.solid', (WidgetTester tester) async {
```

Comment:
Let's add a comment here (as well as before each of the other tests) to indicate that SkiaGold isn't yet able to render shadows in golden tests. Something like:

```
// TODO(garyq): Golden tests currently render shadows as black boxes. We should
// selectively fully render shadows when it is directly relevant.
```

Since the other stuff is tested, we can proceed after adding the TODO comments.

Reply:
Let's just turn on shadows for this test with the flag mentioned by @Piinks.